homeassistant/components/rest/binary_sensor.py

```diff
@@ -142,3 +168,42 @@ def is_on(self):
     def update(self):
         """Get the latest data from REST API and updates the state."""
         self.rest.update()
+        value = self.rest.data
+
+        if self._json_attrs:
+            self._attributes = {}
+            attr = {}
+            if value:
+                try:
+                    if isinstance(self._json_attrs, template.Template):
+                        attr = self._json_attrs.render_with_possible_json_value(value)
```

Comment:
Since a lot of rendering is happening here, it should not happen inside a sync context. We should make this method `aync def async_update(self)` and it should run the rest update in an executor:

```python
await self.hass.async_add_executor_job(self.rest.update)
```

Reply:
Also, remove the fact that it can just be a template. `attr` should only be allowed to be a dictionary.